signer/threshold_validator_test.go

```diff
@@ -173,42 +181,88 @@ func TestThresholdValidator3of3(t *testing.T) {
 	privKeyBytes := privateKey[:]
 	secretShares := tsed25519.DealShares(tsed25519.ExpandSecret(privKeyBytes[:32]), threshold, total)
 
+	tmpDir := t.TempDir()
+
+	var cosigner1, cosigner2, cosigner3 Cosigner
```

Comment:
If there is a way to pull most of this code into a loop, that would help readability of this test.

Reply:
refactored this file to clean up test into a single test body with dynamic test cases for the various threshold/shares combos.